PocketKit/Sources/PocketKit/Collections/CollectionViewController.swift

```diff
@@ -254,46 +265,11 @@ private extension CollectionViewController {
             let margin: CGFloat = environment.traitCollection.shouldUseWideLayout() ? Margins.iPadNormal.rawValue : Margins.normal.rawValue
             let stories = collection.stories?.compactMap { $0 as? CollectionStory } ?? []
 
-            let components = stories.reduce((CGFloat(0), [NSCollectionLayoutItem]())) { result, story in
-                let currentHeight = result.0
-
-                let height = RecommendationCell.fullHeight(
-                    viewModel: model.storyViewModel(for: story),
-                    availableWidth: width - (margin * 2)
-                ) + margin
-
-                var items = result.1
-                let item = NSCollectionLayoutItem(
-                    layoutSize: NSCollectionLayoutSize(
-                        widthDimension: .fractionalWidth(1),
-                        heightDimension: .absolute(height)
-                    )
-                )
-                item.contentInsets = NSDirectionalEdgeInsets(top: 8, leading: 0, bottom: 8, trailing: 0)
-
-                items.append(item)
-
-                return (currentHeight + height, items)
+            if environment.traitCollection.shouldUseWideLayout() {
+                return collectionSectionForWideLayout(with: stories, width: width, margin: margin)
+            } else {
+                return collectionSectionForCompactLayout(with: stories, width: width, margin: margin)
```

Comment:
you could extract this logic in one single method, perhaps called `collectionSection(_:)`

Reply:
I moved this to a separate class under `gridSection`